lib/elixir/lib/kernel/utils.ex

```diff
@@ -22,25 +22,38 @@ defmodule Kernel.Utils do
         _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
       end
 
-    :ok = check_defdelegate_args(args)
-
-    as_args =
-      case append_first and args != [] do
-        true  -> tl(args) ++ [hd(args)]
-        false -> args
+    as_args_list = normalize_args(args)
+    as = Keyword.get(opts, :as, name)
+    Enum.map as_args_list, fn as_args ->
```

Comment:
It compiles right now, but if `Enum` or `List` starts using `defdelegate` the compilation crashes with unpleasant error.